amplify/backend/api/petrakipApi4data/schema.graphql

```diff
@@ -0,0 +1,95 @@
+# General Types and Enums
+
+type S3Object {
+  bucket: String!
+  key: String!
+  region: String!
+}
+
+enum ContentType {
+  image
+  video
+  text
+  audio
+}
+
+type Comment {
+  createdAt: AWSDateTime!
+  content: String!
+}
+
+type OrientationQuestions {
+  question: String!
+}
+
+# Model types -> create DynamoDB tables
+
+type ProfileSettings @model @auth(rules: [{ allow: owner }]) {
+  id: ID!
+  profileImage: S3Object
+  age: Int
+}
+
+type Reflexion
+  @model
+  @auth(
+    rules: [
+      # Defaults to use the "owner" field.
+      { allow: owner }
+      # Authorizes every "email" in sharedUsers[] to read the reflexions
+      { allow: owner, ownerField: "sharedUsers", operations: [read] }
+    ]
+  ) {
+  id: ID!
+  createdAt: AWSDateTime
+  title: String
+  contentType: ContentType
+  content: String
+  asset: S3Object
+  topic: String
+  subTopic: String
+  niveau: String
+  indicators: [String]
+  deleted: Boolean
+  sharedUsers: [String]
+  comments: [Comment]
+  orientationQuestions: [OrientationQuestions]
+  moments: [ReflexionMoment] @connection(keyName: "byReflexion", fields: ["id"])
+}
+
+type Moment
+  @model
+  @auth(
+    rules: [
+      # Defaults to use the "owner" field.
+      { allow: owner }
+      # Authorizes every "email" in sharedUsers[] to read the reflexions
+      { allow: owner, ownerField: "sharedUsers", operations: [read] }
+    ]
+  ) {
+  id: ID!
+  createdAt: AWSDateTime
+  title: String!
+  contentType: ContentType
+  content: String
+  asset: S3Object
+  tags: [String]
+  deleted: Boolean
+  sharedUsers: [String]
+  comments: [Comment]
+  reflexion: [ReflexionMoment] @connection(keyName: "byMoment", fields: ["id"])
+}
+
+# Many-to-Many
+# You can implement many to many using two 1-M @connections, an @key, and a joining @model. For example
+type ReflexionMoment
+  @model(queries: null)
+  @key(name: "byReflexion", fields: ["reflexionID", "momentID"])
+  @key(name: "byMoment", fields: ["momentID", "reflexionID"])
+  @auth(rules: [{ allow: owner }]) {
+  id: ID!
+  reflexionID: ID!
+  momentID: ID!
+  reflexion: Reflexion! @connection(fields: ["reflexionID"])
+  moment: Moment! @connection(fields: ["momentID"])
+}
```

Comment:
@manzanillo it would be easier to discuss if you comment on this line.

> What is the advantage of having a separate relation type for moments referenced in reflections? Otherwise, I would have suggested to handle it just like comments.

As documented [here](https://github.com/oseven-development/petrakip-app/blob/add-data-model-amplify/amplify/README.md#data-model-and-api-structure) with the[ reference to amplify-documentation](https://docs.amplify.aws/lib/datastore/relational/q/platform/js#many-to-many). This is how you handle many-to-many relations. Handling it like comments would make it super complex when a user updates a moment then you would have to search all reflexion etc.....